### PR TITLE
docs: add upgrading info about node service

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: UPGRADING.md [hunk 1]
```diff
@@ -109,6 +109,18 @@ will need to change to accommodate these changes. Most notably:
   longer exported and have been replaced with `node.New` and
   `node.NewDefault` which provide more functional interfaces.
 
+To access any of the functionality previously avalible vai the
```
Comment (cmwaters) [inline UPGRADING.md:112]:
```suggestion
To access any of the functionality previously available via the
```

File: UPGRADING.md [hunk 2]
```diff
@@ -109,6 +109,18 @@ will need to change to accommodate these changes. Most notably:
...
+`node.Node` type, use the `*local.Local` "RPC" client, that exposes
+the full RPC interface provided as direct function calls. Import the
+`github.com/tendermint/tendermint/rpc/client/local` package and pass
+the node service as in the following: 
+
+    node := node.NewDefault() //construct the node object
+    // start and set up the node service 
+
+	client := local.New(node.(local.NodeService))
+	// use client object to interact with the node
```
Comment (cmwaters) [inline UPGRADING.md:122]:
```suggestion
```go
    node := node.NewDefault() //construct the node object
    // start and set up the node service 

    client := local.New(node.(local.NodeService))
    // use client object to interact with the node
```
```

Comment (cmwaters) [reply UPGRADING.md:122]: Lol I can't add the code blocks. Inception doesn't work with markdown

Comment (cmwaters) [reply UPGRADING.md:122]: The main point of this wasn't so much the code blocks (although I think they look better) but more that the indentation was the same

**Comment by cmwaters** [comment]
Also I'm wondering if this change should be targeting master and then get backported to 0.35

File: UPGRADING.md (hunk 2 above)
Comment (tychoish) [reply UPGRADING.md:122]: oh nice.

what is inception?

Comment (cmwaters) [reply UPGRADING.md]: Well recursion really. I was creating a code block for the suggestion that had a code block for the go code inside it. (Which turns out doesn't work)

**Review by cmwaters — APPROVED** [review]
Thanks for that

**Pull request merged**